test/dymaptic.GeoBlazor.Core.Test.ConstructorTesting/Program.cs

```diff
@@ -0,0 +1,254 @@
+
+    static async Task<int> Main(string[] args)
+    {
+        try
+        {
+            // Initialize MSBuild for Roslyn workspace
+            MSBuildLocator.RegisterDefaults();
+
+            // Locate and load the solution
+            var solutionPath = FindSolutionFile(Directory.GetCurrentDirectory());
+            if (string.IsNullOrEmpty(solutionPath))
+            {
+                Console.WriteLine("Could not find solution file (.sln) in the repository.");
+                return 1;
+            }
+
+            // Analyze all projects in the solution
+            var nonCompliantClasses = await AnalyzeSolution(solutionPath);
+
+            // Report results
+            return ReportResults(nonCompliantClasses);
+        }
+        catch (Exception ex)
+        {
+            Console.WriteLine($"Error: {ex.Message}");
+            return 1;
+        }
+    }
+
+    /// <summary>
+    /// Analyzes only the Core project for constructor compliance.
+    /// </summary>
+    static async Task<List<(string className, string filePath)>> AnalyzeSolution(string solutionPath)
+    {
+        using var workspace = MSBuildWorkspace.Create();
+
+        // Load only the Core project instead of entire solution
+        var projectPath = Path.Combine(Path.GetDirectoryName(solutionPath)!, "dymaptic.GeoBlazor.Core", "dymaptic.GeoBlazor.Core.csproj");
+
+        if (!File.Exists(projectPath))
+        {
+            Console.WriteLine($"Could not find Core project at: {projectPath}");
+            return new List<(string className, string filePath)>();
+        }
+
+        Console.WriteLine("Loading dymaptic.GeoBlazor.Core project...");
+        var project = await workspace.OpenProjectAsync(projectPath);
+
+        Console.WriteLine($"Analyzing {project.Name}...");
+        var projectViolations = await AnalyzeProject(project, solutionPath);
+
+        return projectViolations;
+    }
+
+    /// <summary>
+    /// Analyzes a single project for constructor compliance.
+    /// </summary>
+    static async Task<List<(string className, string filePath)>> AnalyzeProject(Project project, string solutionPath)
+    {
+        var violations = new List<(string className, string filePath)>();
+        var compilation = await project.GetCompilationAsync();
+        if (compilation == null) return violations;
+
+        var syntaxTrees = compilation.SyntaxTrees.ToList();
+        var fileCount = syntaxTrees.Count;
+        var currentFile = 0;
+
+        foreach (var syntaxTree in syntaxTrees)
+        {
+            currentFile++;
+            var fileName = Path.GetFileName(syntaxTree.FilePath);
+
+            // Show progress - using carriage return to overwrite the same line
+            Console.Write($"\rAnalyzing file {currentFile}/{fileCount}: {fileName,-50}");
```

Comment:
`\r` is not a standard line break character. Windows is `\r\n` and Unix is `\n`. Typically for cross-platform support we use `$"{Environment.NewLine}"`